### PR TITLE
[CI Fast test stages] Improve stabitility of COMGR tests. Accelerate Dynamic iGemm tests.

### PR DESCRIPTION
**Pull request by atamazov**
- Improves stability of COMGR tests on Jenkins by lowering parallelism (the suspect of instability is HIP runtime).
  - Note: This was prototyped in the [ci-testing-comgr-fix-2](https://github.com/ROCmSoftwarePlatform/MIOpen/tree/ci-testing-comgr-fix-2) branch.
- By-product: Accelerates all Fast tests by replacing Dynamic iGemm tests (>1000 sec which is ridiculous for Fast stages) with their smaller/faster subset (< 10 sec).

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -542,21 +542,26 @@ set(DYNAMIC_IMPLICITGEMM_1X1_ENVS
...
-COMMAND ${DYNAMIC_IMPLICITGEMM_ENVS} $<TARGET_FILE:test_conv2d> --verbose --input 128  128 17 17 --weights 128  128 1 7 --pads_strides_dilations 0 3 1 1 1 1 --disable-backward-data --disable-backward-weights
 
+add_custom_test(test_conv_igemm_dynamic_small
+COMMAND ${DYNAMIC_IMPLICITGEMM_ENVS}     $<TARGET_FILE:test_conv2d> --verbose --input  16  16 56 56 --weights 64  16 1 1 --pads_strides_dilations 0 0 1 1 1 1 --disable-backward-data --disable-backward-weights
+COMMAND ${DYNAMIC_IMPLICITGEMM_ENVS}     $<TARGET_FILE:test_conv2d> --verbose --input  16  64 34 34 --weights 64  64 3 3 --pads_strides_dilations 0 0 1 1 1 1 --disable-backward-data --disable-backward-weights
+COMMAND ${DYNAMIC_IMPLICITGEMM_ENVS}     $<TARGET_FILE:test_conv2d> --verbose --input  32  32 17 17 --weights 32  32 1 7 --pads_strides_dilations 0 3 1 1 1 1 --disable-backward-data --disable-backward-weights
+COMMAND ${DYNAMIC_IMPLICITGEMM_1X1_ENVS} $<TARGET_FILE:test_conv2d> --verbose --input  16 384  8  8 --weights 64 384 1 1 --pads_strides_dilations 0 0 1 1 1 1 --disable-backward-data --disable-backward-weights
+)
+
+add_custom_test(test_conv_igemm_dynamic SKIP_UNLESS_ALL
```
Comment (atamazov) [inline test/CMakeLists.txt:553]: This is original set of tests. Excluded unless "TEST_ALL".

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -542,21 +542,26 @@ set(DYNAMIC_IMPLICITGEMM_1X1_ENVS
...
-COMMAND ${DYNAMIC_IMPLICITGEMM_ENVS} $<TARGET_FILE:test_conv2d> --verbose --input  64  256 34 34 --weights 256  256 3 3 --pads_strides_dilations 0 0 1 1 1 1 --disable-backward-data --disable-backward-weights
-COMMAND ${DYNAMIC_IMPLICITGEMM_ENVS} $<TARGET_FILE:test_conv2d> --verbose --input 128  128 35 35 --weights 128  128 3 3 --pads_strides_dilations 0 0 2 2 1 1 --disable-backward-data --disable-backward-weights
-COMMAND ${DYNAMIC_IMPLICITGEMM_ENVS} $<TARGET_FILE:test_conv2d> --verbose --input  64 1536  8  8 --weights 256 1536 1 1 --pads_strides_dilations 0 0 1 1 1 1 --disable-backward-data --disable-backward-weights
-COMMAND ${DYNAMIC_IMPLICITGEMM_ENVS} $<TARGET_FILE:test_conv2d> --verbose --input 128   48  7  7 --weights 128   48 5 5 --pads_strides_dilations 2 2 1 1 1 1 --disable-backward-data --disable-backward-weights
-COMMAND ${DYNAMIC_IMPLICITGEMM_ENVS} $<TARGET_FILE:test_conv2d> --verbose --input 128  128 17 17 --weights 128  128 1 7 --pads_strides_dilations 0 3 1 1 1 1 --disable-backward-data --disable-backward-weights
 
+add_custom_test(test_conv_igemm_dynamic_small
```
Comment (atamazov) [inline test/CMakeLists.txt:546]: Four configs from the original set where C, N and K divided by 4.

**Comment by carlushuang** [comment]
Hi @atamazov is it because using COMGR to build kernel? Or igemm dynamic run too slow? It is strange that these kernel run several seconds.

**Comment by atamazov** [comment]
> It is strange that these kernel run several seconds.

Not that strange, because all tensors are 16x smaller. And "small list" contains only 4 out of 9 that reside in the "original list". So 1000 / 16 * 4 / 9 = 27 sec. that is close to 10. This rest is due to less cache misses I guess.

**Comment by atamazov** [comment]
@carlushuang 
> is it because using COMGR to build kernel?

What is "it"?

**Comment by carlushuang** [comment]
> @carlushuang
> 
> > is it because using COMGR to build kernel?
> 
> What is "it"?

Oh, just guess around the reason for slow, forget this. Since for each test, it need run both CPU and GPU to cross-validation, the GPU kernel runs very fast even for large tensor, while CPU is very slow for large tensor, right?

**Comment by daniellowell** [comment]
>Oh, just guess around the reason for slow, forget this. Since for each test, it need run both CPU and GPU to cross-validation, the GPU kernel runs very fast even for large tensor, while CPU is very slow for large tensor, right?

Yes, this is right.

**Pull request merged**